core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -984,6 +984,11 @@ class DAGScheduler(
                   job.numFinished += 1
                   // If the whole job has finished, remove it
                   if (job.numFinished == job.numPartitions) {
+                    // Clear failure count for this stage, now that it's succeeded. 
+                    // This ensures that even if subsequent stages fail, triggering 
+                    // a recompute of this stage, we abort because of those failures. 
+                    stage.clearFailures()
```

Comment:
also is that comment supposed to say "... we do not abort ..."?  What do you think of rewording to something like:

We only limit *consecutive* failures of stage attempts, so clear the failure count on success.  This is just in
case this stage is a dependency for *lots* downstream stages, eg. in a long-running streaming app -- we don't
want to penalize the stage if it fails rarely, but its run often enough that its total failure count goes over the limit.

(this isn't perfect either ...)